sled-agent/src/rack_setup/service.rs

```diff
@@ -988,6 +1006,119 @@ impl ServiceInner {
+                .await
+                .map_err(|e| SetupServiceError::DnsResolver(e))?;
+
+            let dpd = DpdClient::new(
+                &format!("http://[{}]:{}", dpd_addr.ip(), dpd_addr.port()),
+                dpd_client::ClientState {
+                    tag: "sled-agent".to_string(),
+                    log: self.log.new(o!(
+                        "component" => "DpdClient"
+                    )),
+                },
+            );
+
+            info!(self.log, "Building Rack Network Configuration");
+            let body = dpd_client::types::Ipv6Entry {
+                addr: BOUNDARY_SERVICES_ADDR.parse().map_err(|e| {
+                    SetupServiceError::BadConfig(format!(
+                        "failed to parse `BOUNDARY_SERVICES_ADDR` as `Ipv6Addr`: {e}"
+                    ))
+                })?,
+                tag: "rss".into(),
+            };
+
+            let mut dpd_port_settings = PortSettings {
+                tag: "rss".into(),
+                links: HashMap::new(),
+                v4_routes: HashMap::new(),
+                v6_routes: HashMap::new(),
+            };
+
+            // TODO handle breakouts
+            // https://github.com/oxidecomputer/omicron/issues/3062
+            let link_id = LinkId(0);
+            let addr: IpAddr = rack_network_config.uplink_ip.parse()
+                .map_err(|e| {
+                    SetupServiceError::BadConfig(format!(
+                    "unable to parse rack_network_config.uplink_up as IpAddr: {e}"))
+                })?;
+
+            let link_settings = LinkSettings {
+                // TODO Allow user to configure link properties
+                // https://github.com/oxidecomputer/omicron/issues/3061
+                params: LinkCreate {
+                    autoneg: false,
+                    kr: false,
+                    fec: PortFec::None,
+                    speed: rack_network_config.uplink_port_speed.clone().into(),
+                },
+                addrs: vec![addr],
+            };
+
+            dpd_port_settings.links.insert(link_id.to_string(), link_settings);
+
+            let port_id: PortId = rack_network_config.uplink_port.parse()
+                .map_err(|e| SetupServiceError::BadConfig(
+                        format!("could not use value provided to rack_network_config.uplink_port as PortID: {e}")
+                ))?;
+
+            let nexthop: Option<Ipv4Addr> = Some(rack_network_config.gateway_ip.parse()
+                .map_err(|e| SetupServiceError::BadConfig(
+                        format!("unable to parse rack_network_config.gateway_ip as Ipv4Addr: {e}")
+                ))?);
+
+            dpd_port_settings.v4_routes.insert(
+                Ipv4Cidr { prefix: "0.0.0.0".parse().unwrap(), prefix_len: 0 }
+                    .to_string(),
+                RouteSettingsV4 { link_id: link_id.0, nexthop },
+            );
+
+            info!(self.log, "Waiting for dendrite to come online");
```

Comment:
Maybe we should print this in the while loop so things are less mysterious if we get stuck here. Printing the error that comes back from `dpd_uptime` would be helpful too.

Reply:
`sled-agent` will now show the following while waiting for `dendrite`:
```
[2023-06-02T02:55:46.031Z]  INFO: SledAgent/19965 on workstation-1: Checking dendrite uptime (component=RSS)
...
[2023-06-02T02:56:01.032Z]  INFO: SledAgent/19965 on workstation-1: Unable to check Dendrite uptime (component=RSS)
    reason: Communication Error: error sending request for url (http://[fd00:1122:3344:101::2]:12224/dpd-uptime): operation timed out
[2023-06-02T02:56:01.032Z]  INFO: SledAgent/19965 on workstation-1: Waiting for dendrite to come online (component=RSS)
```